### PR TITLE
Update ACR Docstrings

### LINKED ISSUE #19571: Review and improve docstrings for ACR as needed.
Until we are generating code from a consistent swagger file, it doesn't make sense to spend much time on docstrings.

Once the swagger and API surface have settled down a bit, loop back and make these docstrings great for customers.

### PR DESCRIPTION
**Pull request by annelo-msft**
Update the docstrings for ACR's public API.

Fixes https://github.com/azure/azure-sdk-for-net/issues/19571

### CODE REVIEW

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/Models/ArtifactArchitecture.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 namespace Azure.Containers.ContainerRegistry
 {
-    /// <summary> The ArtifactArchitecture. </summary>
+    /// <summary> The artifact platform's architecture. </summary>
```
Comment (annelo-msft) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Generated/Models/ArtifactArchitecture.cs:13]: Move docstrings for generated files into swagger.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs [hunk 2]
```diff
@@ -110,6 +110,12 @@ public partial class ArtifactTagProperties
         public string RegistryLoginServer { get { throw null; } }
         public string RepositoryName { get { throw null; } }
     }
+    public static partial class AzureContainerRegistryModelFactory
```
Comment (annelo-msft) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/api/Azure.Containers.ContainerRegistry.netstandard2.0.cs:113]: @AlexanderSher, can you help me understand what changed in the model factory generator that changed the name of this class?  Thanks!

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Artifact/RegistryArtifact.cs [hunk 3]
```diff
@@ -9,7 +9,10 @@
 
 namespace Azure.Containers.ContainerRegistry
 {
-    /// <summary> A helper class that groups information and operations about an image or artifact in this container registry. </summary>
+    /// <summary> `Artifact` is the general term for items stored in a container registry,
+    /// and can include Docker images or other Open Container Initiative (OCI) artifact types.
+    ///
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Artifact/RegistryArtifact.cs:14]: I'm not sure if the spacing translates to the actual docs or not - you may need to do this with the `<para>` tag if you want that kind of formatting.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs [hunk 4]
```diff
@@ -23,7 +23,7 @@ public partial class ContainerRegistryClient
         /// <summary>
         /// Initializes a new instance of the <see cref="ContainerRegistryClient"/> for managing container images and artifacts,
         /// using anonymous access to the registry.  Only operations that support anonymous access are enabled.  Other service
-        /// methods will throw <see cref="RequestFailedException"/> if called.
+        /// methods will throw <see cref="RequestFailedException"/> if called from this client.
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs:26]: Is this also documented in an  `<exception` tag for each operation that exhibits this behavior?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs [hunk 5]
```diff
@@ -93,7 +93,7 @@ protected ContainerRegistryClient()
         /// </summary>
         public virtual Uri Endpoint => _endpoint;
 
-        /// <summary> List repositories in this registry. </summary>
+        /// <summary> List the repositories in this registry. </summary>
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs:96]: nit: 
```suggestion
        /// <summary> List the names of the repositories in this registry. </summary>
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs [hunk 6]
```diff
@@ -17,13 +17,13 @@ public ArtifactManifestProperties()
...
         }
 
-        /// <summary> Delete enabled. </summary>
+        /// <summary> Whether or not this tag can be deleted. </summary>
         public bool? CanDelete { get; set; }
-        /// <summary> Write enabled. </summary>
+        /// <summary> Whether or not this tag can be written to. </summary>
         public bool? CanWrite { get; set;  }
-        /// <summary> List enabled. </summary>
+        /// <summary> Whether or not to include this tag in the collection returned from the tag listing operation. </summary>
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs:24]: Is there a particular method that we can reference here that lists the tags?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs [hunk 7]
```diff
@@ -17,13 +17,13 @@ public ArtifactManifestProperties()
         {
         }
 
-        /// <summary> Delete enabled. </summary>
+        /// <summary> Whether or not this tag can be deleted. </summary>
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs:20]: Does it make sense to add the context like we did for `CanList`?
```suggestion
        /// <summary> Whether or not this tag can be deleted with <see cref="DeleteTag" /> . </summary>
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Repository/ContainerRepository.cs [hunk 8]
```diff
@@ -45,7 +50,7 @@ protected ContainerRepository()
         }
 
         /// <summary>
-        /// Create a new <see cref="RegistryArtifact"/> object for the specified artifact.
+        /// Create a new <see cref="RegistryArtifact"/> helper object for the artifact identified by `tagOrDigest`.
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Repository/ContainerRepository.cs:53]:
```suggestion
        /// Create a new <see cref="RegistryArtifact"/> helper object for the artifact identified by <paramref name="tagOrDigest"/>.
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Repository/ContainerRepository.cs [hunk 9]
```diff
@@ -174,7 +179,7 @@ public virtual async Task<Response> DeleteAsync(CancellationToken cancellationTo
             }
         }
 
-        /// <summary> Delete the repository. </summary>
+        /// <summary> Delete the repository and artifacts that share its name. </summary>
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Repository/ContainerRepository.cs:182]: nit: Is the following also correct? If so, I think this more explicit wording may be a bit more clear.
```suggestion
        /// <summary> Delete the repository and all artifacts that share the name of this repository. </summary>
```

**Review by christothes — APPROVED** [review]

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs (hunk 7 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs:20]: There's non-trivial usage of the Azure CLI for this service, so I'd like to leave it general to that.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs (hunk 6 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Models/ArtifactManifestProperties.cs:24]: Yes, that I can add.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/ContainerRegistryClient.cs:26]: Technically yes, as I did a pass to add these whereever the method makes a service call.  The service is changing which endpoints allow anonymous access, so we decided not to indicate it method-by-method.

**Pull request merged**